upgrade-2.x.sh

```diff
@@ -337,6 +340,38 @@ function pre_update_fix_bootfiles_hook {
     mount --bind "$bootfiles_temp"  /etc/hostapp-update-hooks.d/0-bootfiles
 }
 
+#######################################
+# Updat problematic persistent logging env var
+# Earlier supervisors might have set it to "", and
+# that doesn't validate on newer supervisor versions.
+# Convert into proper false value.
+# Globals:
+#   API_ENDPOINT
+#   APIKEY
+#   CONFIGJSON
+#   DEVICEID
+# Returns:
+#   None
+#######################################
+function persistent_logging_config_var {
+    PROBLEMATIC_ENV_VAR=$(curl --retry 10 --silent -X GET "${API_ENDPOINT}/v5/device_config_variable?\$filter=device%20eq%20${DEVICEID}" -H "Content-Type: application/json" -H "Authorization: Bearer ${APIKEY}" | jq -r '.d[] | select((.name == "RESIN_SUPERVISOR_PERSISTENT_LOGGING") and (.value == "")) | .id')
+    if [ -n "${PROBLEMATIC_ENV_VAR}" ]; then
+        local tmpfile
+        log "Updating problematic RESIN_SUPERVISOR_PERSISTENT_LOGGING config variable"
+        curl --retry 10 --silent -X PATCH \
+            "${API_ENDPOINT}/v5/device_config_variable(${PROBLEMATIC_ENV_VAR})" \
+            -H "Content-Type: application/json" \
+            -H "Authorization: Bearer ${APIKEY}" \
+            --data '{
+                "value": "false"
+            }' >> /dev/null
+        log "Updating config.json with sanitized '.persistentLogging' value."
+        tmpfile=$(mktemp -t configjson.XXXXXXXX)
+        jq '.persistentLogging="false"' < "${CONFIGJSON}" > "${tmpfile}"
+        cp "${tmpfile}" "$CONFIGJSON" && sync
```

Comment:
Shouldn't this be a rename (mv)?

Reply:
@agherzan much difference in this case? Since it's a copy anyways (from memory on `/tmp` to SD card on `/mnt/boot`), I think it always needs a copy.

Reply:
I've updated, but the question stands.

Reply:
I was pointing at the atomicity of the operation. Missed the fact that they are on different fs but I think we need to make this operation atomic.